### PR TITLE
added shipping details

### LINKED ISSUE #1725: Cart shipping functionality
When the cart is empty then shipping charges will not show.
if the total value will be greater or equal to 500 then shipping will free other wise add 40 rupees for shipping charges.

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**
fixes #1725

## Screenshots

Screenshot 2024-07-09 083901
Screenshot 2024-07-09 083936
Screenshot 2024-07-09 083958

## Summary by CodeRabbit

- **New Features**
  - Introduced the `OrderSummary` component to calculate and display subtotal, shipping cost, and total amount for an order.
  - Integrated currency formatting in Indian Rupees (INR) for order amounts.

- **Enhancements**
  - Replaced static order summaries with the dynamic `OrderSummary` component in Cart, Checkout, and Payment pages.
  - Updated the `Pay ₹1100` button text to `Proceed to payment` for better clarity.

### CODE REVIEW

**Comment by DharshiBalasubramaniyam** [comment]
@codervivek5 can you please merge the pr?

File: src/components/Order/OrderSummary.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    const items = useSelector(state => state.cart.items)
+
+    const itemsTotal = items.reduce((acc, item) => acc + item.total, 0);
+
+    const shippingThreshold = 500.00;
+    const shippingRate = 40.00;
+
+    let shipping = itemsTotal >= shippingThreshold ? 0.00 : shippingRate;
+    let total = itemsTotal + shipping;
```
[the remark this answers is not in the document]
Comment (codervivek5) [reply src/components/Order/OrderSummary.jsx]: check this code.

File: src/components/Order/OrderSummary.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    if (itemsTotal === 0) {
+        return <></>;
+    }
```
[the remark this answers is not in the document]
Comment (codervivek5) [reply src/components/Order/OrderSummary.jsx]: also check this code.

**Comment by codervivek5** [comment]
@DharshiBalasubramaniyam kindly take a look of code rabbit code

**Comment by DharshiBalasubramaniyam** [comment]
> @DharshiBalasubramaniyam kindly take a look of code rabbit code

@codervivek5 done!

**Comment by codervivek5** [comment]
button clicks are not working.
- The checkout button not working in cart page
image

- Proceed to Payment button not working in checkout page
image

**Comment by DharshiBalasubramaniyam** [comment]
> button clicks are not working.
> 
> * The checkout button not working in cart page
>   image
> * Proceed to Payment button not working in checkout page
>   image

I have not checked those things because these are not related to my issue.

If you prefer I can create a new pr for this. You do not want add label to that pr.

**Pull request merged**